lib/facter/consul_version.rb

```diff
@@ -0,0 +1,17 @@
+# consul_version.rb
+
+Facter.add(:consul_version) do
+  confine :kernel => 'Linux'
+  setcode do
+    Facter::Core::Execution.exec('consul --version | head -1 | cut -f2 -d\ | cut -c2-')
```

Comment:
Can you do the raw exec and do the string manipulation in ruby? Same with powershell.